This is not needed. Gentoo automatically collect license files into
the documentation directory.

diff '--color=auto' -urp rocm-cmake-rocm-6.1.2.orig/share/rocmcmakebuildtools/cmake/ROCMCreatePackage.cmake rocm-cmake-rocm-6.1.2/share/rocmcmakebuildtools/cmake/ROCMCreatePackage.cmake
--- rocm-cmake-rocm-6.1.2.orig/share/rocmcmakebuildtools/cmake/ROCMCreatePackage.cmake	2024-07-02 02:49:26.039660059 -0700
+++ rocm-cmake-rocm-6.1.2/share/rocmcmakebuildtools/cmake/ROCMCreatePackage.cmake	2024-07-02 02:50:08.135074401 -0700
@@ -409,7 +409,6 @@ macro(rocm_create_package)
     if(PARSE_PTH)
         rocm_parse_python_syspath(${LIB_DIR} ${PARSE_NAME})
     endif()
-    rocm_setup_license(${PARSE_HEADER_ONLY})
     if(PARSE_COMPONENTS)
         rocm_set_comp_cpackvar(PARSE_HEADER_ONLY "${PARSE_COMPONENTS}")
     endif()
